tests/beaker_tests/cisco_bgp_neighbor_af/test_bgpneighboraf.rb

```diff
@@ -539,154 +603,170 @@ def build_manifest_bgp_nbr_af(tests, id)
 
 # Wrapper for bgp_nbr_af specific settings prior to calling the
 # common test_harness.
-def test_harness_bgp_nbr_af(tests, id)
+def test_harness_bgp_nbr_af(tests, id, platform)
   af = af_title_pattern_munge(tests, id, 'bgp_neighbor_af')
   logger.info("\n--------\nTest Case Address-Family ID: #{af}")
 
-  # Set up remote-as if necessary
-  bgp_nbr_remote_as(agent, tests[id][:remote_as]) if tests[id][:remote_as]
-
   tests[id][:ensure] = :present if tests[id][:ensure].nil?
-  tests[id][:show_pattern] = af_pattern(tests, id, af)
   tests[id][:resource_cmd] = puppet_resource_cmd(af)
 
-  # Build the manifest for this test
-  build_manifest_bgp_nbr_af(tests, id)
+  if platform == 'ios_xr' && af[:vrf] != 'default'
+    logger.info("\n--------\nSkip Case Address-Family ID: #{af} for ios_xr")
+  else
+    # Build the manifest for this test
+    build_manifest_bgp_nbr_af(tests, id, af, platform)
 
-  test_harness_common(tests, id)
-  tests[id][:ensure] = nil
+    test_harness_common(tests, id)
+    tests[id][:ensure] = nil
+  end
 end
 
 #################################################################
 # TEST CASE EXECUTION
 #################################################################
 test_name "TestCase :: #{testheader}" do
+  platform = fact_on(agent, 'os.name')
   # -------------------------------------------------------------------
   logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
-  node_feature_cleanup(agent, 'bgp')
+  init_bgp(master, agent)
 
   # -----------------------------------
   id = 'default_properties'
-  test_harness_bgp_nbr_af(tests, id)
+  test_harness_bgp_nbr_af(tests, id, platform)
 
   tests[id][:ensure] = :absent
-  test_harness_bgp_nbr_af(tests, id)
+  test_harness_bgp_nbr_af(tests, id, platform)
+  cleanup_bgp(master, agent)
 
   # -------------------------------------------------------------------
   logger.info("\n#{'-' * 60}\nSection 2. Non Default Property Testing")
-  node_feature_cleanup(agent, 'bgp')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_A1')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_A2')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_A3')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_D')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_M')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_N')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_S1')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_S2')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_S3')
-  test_harness_bgp_nbr_af(tests, 'non_default_properties_W')
+  init_bgp(master, agent)
```

Comment:
If you're calling init_bgp here, no need to call cleanup_bgp right before it (init does a cleanup and an init)